components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -6574,4 +6574,276 @@ private void validateAndUpdateURITemplates(API api, int tenantId) throws APIMana
             }
         }
     }
+
+    /**
+     * Add a new gateway policy mapping to the database.
+     *
+     * @param gatewayGlobalPoliciesList List of Gateway Policy objects
+     * @param description               Description of the policy mapping
+     * @param name                      Name of the policy mapping
+     * @param orgId                     Organization ID
+     * @return UUID of the added policy mapping
+     * @throws APIManagementException
+     */
+    public String applyGatewayGlobalPolicies(List<OperationPolicy> gatewayGlobalPoliciesList, String description,
+            String name, String orgId) throws APIManagementException {
+        String policyMappingUUID = UUID.randomUUID().toString();
+        return apiMgtDAO.addGatewayGlobalPolicy(gatewayGlobalPoliciesList, description, name, orgId, policyMappingUUID);
+    }
+
+    /**
+     * Add a new gateway policy mapping deployment to the database.
+     *
+     * @param gatewayPolicyDeploymentMap Policy deployment status to gateway labels map
+     * @param orgId                      Organization ID
+     * @param gatewayPolicyMappingId     UUID of the policy mapping
+     * @throws APIManagementException
+     */
+    public void engageGatewayGlobalPolicies(Map<Boolean, List<GatewayPolicyDeployment>> gatewayPolicyDeploymentMap,
+            String orgId, String gatewayPolicyMappingId) throws APIManagementException {
+
+        APIGatewayManager gatewayManager = APIGatewayManager.getInstance();
+        List<GatewayPolicyDeployment> gatewayPolicyDeploymentList = new ArrayList<>();
+        List<GatewayPolicyDeployment> gatewayPolicyUndeploymentList = new ArrayList<>();
+        Map<String, Set<String>> gatewaysToAdd = new HashMap<>();
+        Map<String, Set<String>> gatewaysToRemove = new HashMap<>();
+
+        for (boolean isDeployment : gatewayPolicyDeploymentMap.keySet()) {
```

Comment:
can `gatewayPolicyDeploymentMap` be null in any case?

Reply:
The gatewayPolicyDeploymentMap will never be null under any circumstance. This is ensured during the initial assignment of values to the gatewayPolicyDeploymentMap, where validation occurs to check if the OperationPolicyData is empty in GatewayPoliciesApiServiceImpl. If it is not empty, the values are obtained from it using the method GatewayPolicyMappingUtil.fromDTOToGatewayPolicyDeploymentMap. However, in cases where there are no new values assigned for the gateway deployments, it will return an empty ArrayList object.